torch/csrc/jit/tensorexpr/bounds_inference.h

```diff
@@ -28,11 +28,28 @@ using BoundsInfo =
 
 TORCH_API BoundsInfo inferBounds(Stmt* s, bool distinctAccessKinds = true);
 
+// Bounds inference caching the analysis. The MemDependencyChecker must already
+// have been run.
+TORCH_API BoundsInfo getInferredBounds(
+    analysis::MemDependencyChecker& analyzer,
+    Stmt* s,
+    bool distinctAccessKinds = true);
+
 TORCH_API void printBoundsInfo(const BoundsInfo& v);
 
 TORCH_API std::vector<const Expr*> getBoundExtents(
     const std::vector<TensorAccessBoundsInfo>& infos);
 
+// The kind of dependency found, in increasing order of exclusivity.
+enum class HazardKind {
```

Comment:
Why call it a hazard? Isn't it just another kind of dependency? I can see that some of these could prevent transformations like loop fusion, but that still doesn't make them a hazard, IMO.

Reply:
I pretty much consider only RAW as a real dependency, and the other two kinds as other things. Wikipedia kind of backs me up here: https://en.wikipedia.org/wiki/Hazard_(computer_architecture) :
```
* read after write (RAW), a true dependency
* write after read (WAR), an anti-dependency
* write after write (WAW), an output dependency
```

Basically I just wanted to distinguish them from the MemDepChecker's depedency chains, because this code doesn't account for anything between the two Stmts provided as inputs.